Adding gear output to the JSON reporting

pontifechs opened the pull request:
The HTML report already has this, so it'd be nice if the JSON had it as well.

pontifechs committed:
Adding Gear output to the JSON reporting (similar to the HTML reporting)

pontifechs committed:
JSON reporting cleanup

navv1234 commented:
Change the "thing" to "options" or something. Also if you want to do this properly, it might actually make sense to change the "thing" to an "options" object, and put the field information from item_t::parsed_input_t there.

pontifechs committed:
Changing the json key for the encoded_item line.

pontifechs commented:
Yeah, I couldn't quite come up with a good name for it, so I just put in thing and moved on. Forgot to come back and think about the name. I think just "encoded_item" works pretty well. 

As far as splitting it out into an object, that sort of defeats the purpose I had in mind. The original thought was to be able to recreate that item exactly in another sim execution. If I split it out into the individual components, now the consumer has to be able to reassemble them to be able to send it back to simc.
